resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala

```diff
@@ -158,17 +158,21 @@ private[spark] class KubernetesSparkRestServer(
                 handleError("Unauthorized to submit application.")
               } else {
                 val tempDir = Utils.createTempDir()
-                val appResourcePath = resolvedAppResource(appResource, tempDir)
+                val appResourcePath = resolveAppResourceToLocalPath(appResource, tempDir)
                 val writtenJars = writeUploadedJars(uploadedJars, tempDir)
                 val writtenFiles = writeUploadedFiles(uploadedFiles)
                 val resolvedSparkProperties = new mutable.HashMap[String, String]
                 resolvedSparkProperties ++= sparkProperties
 
                 // Resolve driver classpath and jars
-                val originalJars = sparkProperties.get("spark.jars")
+                val nonUploadedJars = sparkProperties.get("spark.jars")
                   .map(_.split(","))
+                  .map(_.filter(Utils.resolveURI(_).getScheme match {
+                    case "file" | null => false
+                    case _ => true
+                  }))
                   .getOrElse(Array.empty[String])
-                val resolvedJars = writtenJars ++ originalJars ++ Array(appResourcePath)
+                val resolvedJars = writtenJars ++ nonUploadedJars ++ Array(appResourcePath)
```

Comment:
Here we're missing adding jars with the scheme `local` to the driver's classpath. Note that while we want to add said jars to the classpath using just the raw path, they should still be put in `spark.jars` with the full URI so that executors pick them up from their local disks, as opposed to having the driver upload them.